### ISSUE TITLE
GRAPH-BERT: Only Attention is Needed for Learning Graph Representations

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#276]
* based on the attention mechanism without any graph convolution or aggregation operators
* train GRAPH-BERT with sampled linkless subgraphs within their local contexts
* fine-tune GRAPH-BERT on node classification and graph clustering tasks specifically
* Architecture of the GRAPH-BERT Model
  <image src="https://user-images.githubusercontent.com/67627471/188317998-dd632093-ba03-468f-82ed-6fb17494d629.png" width=80% height=80%>
* (1) linkless subgraph batching
  * doesn’t require any label supervision information actually, but partial of the labels will be used for the fine-tuning application task on node classification to be introduced later
  * Instead of working on the complete graph G, GRAPH-BERT will be trained with linkless subgraph batches sampled from the input graph instead
  * --> effectively enable the learning of GRAPH-BERT to parallelize on extremely large-sized graphs that the existing graph neural networks cannot handle
  * introduce the top-k intimacy sampling approach
* (2) node input embedding
  * doesn’t require any node orders of the input sampled sub-graph
  * propose to serialize the input subgraph nodes into certain ordered list
* (3) graph-transformer based encoder
  * update the nodes’ representations iteratively with multiple layers (D layers)
  * aim to apply the graph-transformer to get the representations of the target node only
* Learning Performance of GRAPH-BERT (node classification)
  <image src="https://user-images.githubusercontent.com/67627471/188318236-0b6b82a8-85ae-4843-91bc-de05e1fac000.png" width=40% height=40%>
